jax/core.py

```diff
@@ -91,13 +91,10 @@
   _check_jaxpr as _check_jaxpr,
   _check_map as _check_map,
   _compact_eqn_should_include as _compact_eqn_should_include,
-  _dim_handler_and_canonical as _dim_handler_and_canonical,
```

Comment:
How is this related to the rest of the PR?

Reply:
These are just drive-by cleanup of the jax.core namespace, for some names related to shape polymorphism